packages/fxa-settings/src/models/reliers/oauth-relier.ts

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ContextValidation as V, bind, ContextKeyTransforms as T } from "../../lib/context";
+import { Context } from "../../lib/context/interfaces/context";
+import { RelierData, Relier } from "./rellier";
+
+/**
+ * A convenience interface for easy mocking / testing. Use this type in components.
+ */
+export interface OAuthRelierData extends RelierData {
+  keysJwk: string | undefined;
+  // permissions are individual scopes
+  permissions: string | undefined;
+  // redirectTo is for future use by the oauth flow. redirectTo
+  // would have redirectUri as its base.
+  redirectTo: string | undefined;
+  // Max age since last auth
+  maxAge: string | undefined;
+}
+
+export class OAuthRelier extends Relier implements OAuthRelierData
+{
+  @bind([V.isString], T.snakeCase)
```

Comment:
The T.snakeCase lets the binding know that the context holds this value under a key with snake case, i.e. keys_jkw.

Reply:
Super nice 👌